server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -973,6 +973,14 @@ public List<? extends CoordinatorDuty> getDuties()
     {
       return duties;
     }
+
+    @Override
+    public String toString()
+    {
+      return "DutiesRunnable{" +
+             "dutiesRunnableAlias='" + dutiesRunnableAlias + '\'' +
+             '}';
+    }
```

Comment:
Oh, yeah, the `VisibleForTesting` is ubiquitous 😅

It would be good to pull out `DutiesRunnable`. Right now, it seems to be directly using pretty much all the fields that `DruidCoordinator` contains. That's probably why it is still hanging around here and why it is not a _static_ inner class either.

The preferable way to do this would be for `DruidCoordinator` to expose a bunch of methods to update the state of segmentManager or other things that `DutiesRunnable` needs to access. And the `DutiesRunnable` constructor just gets the `DruidCoordinator` instance. `DruidCoordinator` already exposes a few of these utility methods such as `moveSegment()` or `markSegmentAsUnused()` which are
used by the actual duties themselves.

Let me know if this approach makes sense. We can get it done in a follow-up PR.